RGLServerPlugin/src/LidarPatternLoader.cc

```diff
@@ -213,6 +214,37 @@ bool LidarPatternLoader::LoadPatternFromPresetPath(const sdf::ElementConstPtr& s
     return true;
 }
 
+bool LidarPatternLoader::LoadPatternFromLidar2d(const sdf::ElementConstPtr& sdf, std::vector<rgl_mat3x4f>& outPattern)
+{
+    if (!sdf->HasElement("horizontal")) {
+        ignerr << "Failed to load uniform pattern. A horizontal element is required, but it is not set.\n";
+        return false;
+    }
+
+    ignition::math::Angle hMin, hMax;
+    int hSamples;
+
+    if (!LoadAnglesAndSamplesElement(sdf->FindElement("horizontal"), hMin, hMax, hSamples)) {
+        return false;
+    }
+
+    outPattern.reserve(hSamples);
+
+    ignition::math::Angle hStep((hMax - hMin) / static_cast<double>(hSamples));
+
+    auto hAngle = hMin;
+    for (int i = 0; i < hSamples; ++i) {
+        outPattern.push_back(
+            AnglesToRglMat3x4f(ignition::math::Angle::Zero,
+                                // Inverse and shift 90deg pitch to match uniform pattern from Gazebo
+                                ignition::math::Angle::HalfPi,
+                                hAngle));
+        hAngle += hStep;
+    }
```

Comment:
If the `outPattern` comes not empty, the new rays will be added to the old ones. I suggest clearing the vector before reserving it. I saw the same bug occur in the other patterns also.